code/clue.js

```diff
@@ -1,59 +1,284 @@
+  "Colonel Mustard",
+  "Mrs White",
+];
+
+//array of objects
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol,
+];
 
-const weapons = []
+//array of strings
+const weaponsArray = [
+  "Rope",
+  "Knife",
+  "Candlestick",
+  "Dumbbell",
+  "Poison",
+  "Axe",
+  "Bat",
+  "Trophy",
+  "Pistol",
+];
 
-const rooms = []
+//array of strings
+const favouriteWeapon = [
+  "ropE",
+  "axE",
+  "poisoN",
+  "baT",
+  "candlesticK",
+  "dumbbelL",
+];
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+//array of strings
+const rooms = [
+  "Dining Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+// function randomly selects one item from the array you pass in to the function
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
-// This function will be invoked when you click on the killer card.
+//an object that keeps the mystery
+const mysteryObject = {
+  killer: null,
+  weapon: null,
+  room: null,
+};
+
+//function invoked when you click on the killer card
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  //randomly select a killer from the suspects + add that to the mystery object.
+  mysteryObject.killer = randomSelector(suspects); //random array object will be saved into an object called mystery.killer
+
+  //change the background color of the card to the one connected to the chosen killer
+  document.getElementById("killerCard").style.background =
+    mysteryObject.killer.color;
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
+  //show the full name + other info of the killer
   document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+    "killerName"
+  ).innerHTML = `${mysteryObject.killer.firstName} ${mysteryObject.killer.lastName}`;
+  document.getElementById(
+    "killerAge"
+  ).innerHTML = `${mysteryObject.killer.age} years`;
+  document.getElementById("killerImage").src = `${mysteryObject.killer.image}`;
+  document.getElementById("killerOccupation").innerHTML =
+    mysteryObject.killer.occupation;
```

Comment:
On some you have `${}`, like in .innerHTML = `${mysteryObject.killer.age} years`;
and in other you have not the `${}`, like in: .innerHTML =mysteryObject.killer.occupation;
Maybe you should have the same on every, having the `${}` on all, or no `${}` at all... Or maybe there is a reason which I missing something out here? Please tell me, so I learn :)

Reply:
We should use template strings `${}` if we want to pass both variable and fixed string, like
`'${myVariable} - this is my variable value'`
If we want to pass only variable, we can omit `${}`, like `myVariable`.
In Rebeka's code they are used properly :)